[DOC] New content: New web page for visualize section (#1402)
opensearch-project/documentation-website: _dashboards/visualize/viz-index.md

---
layout: default
title: Visualize
nav_order: 4
has_children: true
---

# Visualize

Data visualization is representing your data in a visual context. By visualizing your data, you translate complex, high-volume, or numerical data into a visual representation that is easier to process.

> OpenSearch Dashboards gives you data visualization tools to improve and automate the visual communication process. Using visual elements like charts, graphs, or maps to represent data, gives you the tools to advance business intelligence and support data-driven decision-making and strategic planning.

Reviewer: ```suggestion
OpenSearch Dashboards gives you data visualization tools to improve and automate the visual communication process. Using visual elements like charts, graphs, or maps to represent data gives you the tools to advance business intelligence and support data-driven decision-making and strategic planning.
```

# Visualization types in OpenSearch Dashboards

Dashboards has several visualization types to support your data analysis needs. The following sections give you an overview of the visualization types in Dashboards and their common use cases.

## Area charts:  Basic and stacked

 <img src="{{site.url}}{{site.baseurl}}/images/area-chart-1.png" alt="Example area chart in OpenSearch Dashboards" width="60%" height="30%">

> Area charts depict changes over time, and they are commonly used to show trends. Area charts more efficiently identify patterns in log data,such as sales data for a time range and trends over that time. Learn how to create and use [Area charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/area-charts) in Dashboards.

Reviewer 2: nit: add space after comma: "log data,such as" -> "log data, such as"

## Bar charts

<img src="{{site.url}}{{site.baseurl}}/images/bar-chart-1.png" alt="Example vertical bar chart in OpenSearch Dashboards" width="60%" height="30%">

<img src="{{site.url}}{{site.baseurl}}/images/bar-horizontal-1.png" alt="Example horizontal bar chart in OpenSearch Dashboards" width="60%" height="30%">

> Bar charts, vertical or horizontal, compare categorical data and demonstrate changes of a variable over a period of time. Learn how to create and use [Bar charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/bar-charts) in Dashboards.

Reviewer: Should this be something like "Reference [Bar charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/bar-charts)
 to learn how to create and use Bar charts in Dashboards." ?

## Data tables

<img src="{{site.url}}{{site.baseurl}}/images/data-table-1.png" alt="Example data table in OpenSearch Dashboards" width="75%" height="35%">

> Data tables, or tables, show your raw data in tabular form. Learn how to create and use [Data tables]({{site.url}}{{site.baseurl}}/dashboards/visualize/tables) in Dashboards.

Reviewer: Same as previous comment

## Gantt chart

<img src="{{site.url}}{{site.baseurl}}/images/gantt-chart.png" alt="Example Gantt chart in OpenSearch Dashboards" width="75%" height="35%">

> Gantt charts show the start, end, and duration of unique events in a sequence. Gantt charts are useful in trace analytics, telemetry, and anomaly detection use cases, where you want to understand interactions and dependencies between various events in a schedule. Learn how to create and use [Gantt charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/gantt/) in Dashboards.

Reviewer: Should this be something like "Reference [Gantt charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/gantt/) to learn how to create and use Gnatt charts in Dashboards." ?

> Gantt charts show the start, end, and duration of unique events in a sequence. Gantt charts are useful in trace analytics, telemetry, and anomaly detection use cases, where you want to understand interactions and dependencies between various events in a schedule. Learn how to create and use [Gantt charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/gantt/) in Dashboards.

Reviewer: ```suggestion
Gantt charts show the start, end, and duration of unique events in a sequence. Gantt charts are useful in trace analytics, telemetry, and anomaly detection use cases where you want to understand interactions and dependencies between various events in a schedule. Learn how to create and use [Gantt charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/gantt/) in Dashboards.
```

> Gantt charts show the start, end, and duration of unique events in a sequence. Gantt charts are useful in trace analytics, telemetry, and anomaly detection use cases, where you want to understand interactions and dependencies between various events in a schedule. Learn how to create and use [Gantt charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/gantt/) in Dashboards.

Reviewer: ```suggestion
Gantt charts show the start, end, and duration of unique events in a sequence. Gantt charts are useful in trace analytics, telemetry, and anomaly detection use cases where you want to understand interactions and dependencies between various events in a schedule. Learn how to create and use [Gantt charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/gantt/) in Dashboards.
```

## Gauge charts

<img src="{{site.url}}{{site.baseurl}}/images/gauge-chart-1.png" alt="Example gauge chart in OpenSearch Dashboards" width="60%" height="30%">

> A gauge chart displays how much there is of the thing you are measuring. In a gauge chart, this measurement can exist alone or in relation to another measurement, such as comparing actual sales to the sales goal. Learn how to create and use [Gauge charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/gauge) in Dashboards.

Reviewer: Should this be something like "Reference [Gauge charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/gauge) to learn how to create and use Gauge charts in Dashboards." ?

## Heat maps

<img src="{{site.url}}{{site.baseurl}}/images/heatmap-1.png" alt="Example heat map in OpenSearch Dashboards" width="60%" height="30%">

A heat map is a view of a histogram (a graphical representation of the distribution of numerical data) over time. Instead of using bar height as a representation of frequency, as with a histogram, heat maps use cells, coloring a cell proportional to the number of values. Learn how to create and use [Heat maps]({{site.url}}{{site.baseurl}}/dashboards/visualize/heat-maps) in Dashboards.
## Line graphs

<img src="{{site.url}}{{site.baseurl}}/images/line-graph-1.png" alt="Example line graph in OpenSearch Dashboards" width="60%" height="30%">

> Line charts compare changes in measure values over period of time, such as gross sales by month or gross sales and net sales by month. Learn how to create and use [Line charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/line-charts) in Dashboards.

Reviewer: Same as previous comment

> Line charts compare changes in measure values over period of time, such as gross sales by month or gross sales and net sales by month. Learn how to create and use [Line charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/line-charts) in Dashboards.

Reviewer: Should this be something like "Reference [Line charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/line-charts) to learn how to create and use Line charts in Dashboards." ?

## Metric charts

<img src="{{site.url}}{{site.baseurl}}/images/metric-chart-1.png" alt="Example metric chart in OpenSearch Dashboards" width="60%" height="30%">

> Metric charts show a numerical value, such as a key performance indicator (KPI), to visualize a comparison between a key value and its target value. Metric charts display a value comparison, the two values being compared, and a progress bar. Learn how to create and use [Metric charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/metric-charts) in Dashboards.

Reviewer: Same as previous comment

## Maps
### Coordinate maps

<img src="{{site.url}}{{site.baseurl}}/images/map-1.png" alt="Example map in OpenSearch Dashboards" width="60%" height="30%">

> Coordinate maps show location-based data on a map. Use coordinate maps to visualize GPS data (latitude and longitude coordinates) on a map.Learn how to create and use [Coordinate maps]({{site.url}}{{site.baseurl}}/dashboards/visualize/coordinate-maps) in Dashboards.

Reviewer: ```suggestion
Coordinate maps show location-based data on a map. Use coordinate maps to visualize GPS data (latitude and longitude coordinates) on a map. Learn how to create and use [Coordinate maps]({{site.url}}{{site.baseurl}}/dashboards/visualize/coordinate-maps) in Dashboards.
```

For information about OpenSearch-supported coordinate field types, see [Geographic field types]({{site.url}}{{site.baseurl}}/opensearch/supported-field-types/geo-shape/) and [Cartesian field types]({{site.url}}{{site.baseurl}}/opensearch/supported-field-types/xy/). 

### Region maps

Region maps show patterns and trends across geographic locations. A region map is one of the basemaps in Dashboards. Learn how to create and use [Region maps]({{site.url}}{{site.baseurl}}/dashboards/visualize/geojson-regionmaps/) in Dashboards. 

## Pie charts

<img src="{{site.url}}{{site.baseurl}}/images/pie-chart-1.png" alt="Example pie chart in OpenSearch Dashboards" width="60%" height="30%">

> Pie charts compare values for items in a dimension, such as  a percentage of a total amount. Learn how to create and use [Pie charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/pie-charts/) in Dashboards.

Reviewer: Should this be moved up to below the heading?

> Pie charts compare values for items in a dimension, such as  a percentage of a total amount. Learn how to create and use [Pie charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/pie-charts/) in Dashboards.

Reviewer: Should this be something like "Reference [Pie charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/pie-charts/) to learn how to create and use Pie charts in Dashboards." ?

## TSVB

<img src="{{site.url}}{{site.baseurl}}/images/TSVB-1.png" alt="Example TSVB in OpenSearch Dashboards" width="60%" height="30%">

> Time-series visual builder provides a detailed time series visualization. For example, you can use TSVB to show data over time, such as flights by status over time or flight delays by delay type over time. Learn how to create and use [TSVB]({{site.url}}{{site.baseurl}}/dashboards/visualize/TSVB/) in Dashboards. 

Reviewer: Should this be something like "Reference [TSVB]({{site.url}}{{site.baseurl}}/dashboards/visualize/TSVB/) to learn how to create and use TSVB in Dashboards." ?

> Time-series visual builder provides a detailed time series visualization. For example, you can use TSVB to show data over time, such as flights by status over time or flight delays by delay type over time. Learn how to create and use [TSVB]({{site.url}}{{site.baseurl}}/dashboards/visualize/TSVB/) in Dashboards. 

Reviewer: ```suggestion
Time-series visual builder provides a detailed time series visualization. For example, you can use TSVB to show data over time, such as flights by status over time, or flight delays by delay type over time. Learn how to create and use [TSVB]({{site.url}}{{site.baseurl}}/dashboards/visualize/TSVB/) in Dashboards. 
```

## Tag cloud

<img src="{{site.url}}{{site.baseurl}}/images/word-cloud-1.png" alt="Example Tag cloud in OpenSearch Dashboards" width="60%" height="30%">

> Tag (or word) clouds are a way to display how often a word is used in relation to other words in a dataset. The best use for this type of visual is to show word or phrase frequency. Learn how to create and use [Tag clouds]({{site.url}}{{site.baseurl}}/dashboards/visualize/tag-clouds/) in Dashboards.

Reviewer: Same as comment above.

> Tag (or word) clouds are a way to display how often a word is used in relation to other words in a dataset. The best use for this type of visual is to show word or phrase frequency. Learn how to create and use [Tag clouds]({{site.url}}{{site.baseurl}}/dashboards/visualize/tag-clouds/) in Dashboards.

Reviewer: Should this be something like "Reference [Tag clouds]({{site.url}}{{site.baseurl}}/dashboards/visualize/tag-clouds/) to learn how to create and use Tag clouds in Dashboards." ?

## Timeline

<img src="{{site.url}}{{site.baseurl}}/images/timeline-1.png" alt="Example Timeline in OpenSearch Dashboards" width="60%" height="30%">

> Timelines  show the occurrence of events in chronological order, allowing  you to see when things occur and how they change over time. Learn how to create and use [Timelines]({{site.url}}{{site.baseurl}}/dashboards/visualize/timeline/) in Dashboards.

Reviewer: Same as comment above.

> Timelines  show the occurrence of events in chronological order, allowing  you to see when things occur and how they change over time. Learn how to create and use [Timelines]({{site.url}}{{site.baseurl}}/dashboards/visualize/timeline/) in Dashboards.

Reviewer: Should this be something like "Reference [Timelines]({{site.url}}{{site.baseurl}}/dashboards/visualize/timeline/) to learn how to create and use Timelines in Dashboards." ?

> Timelines  show the occurrence of events in chronological order, allowing  you to see when things occur and how they change over time. Learn how to create and use [Timelines]({{site.url}}{{site.baseurl}}/dashboards/visualize/timeline/) in Dashboards.

Author: Good word choice catch!

## VisBuilder

<img src="{{site.url}}{{site.baseurl}}/images/drag-drop-generated-viz.png" alt="Example Timeline in OpenSearch Dashboards" width="60%" height="30%">

> VisBuilder is used to create data visualizations with a drag-and-drop gesture. It gives you an immediate view of your data without the need to preselect the visualization output. Learn how to create and use [VisBuilder]({{site.url}}{{site.baseurl}}/dashboards/visbuilder/) in Dashboards.

Reviewer: Same as comment above.

> VisBuilder is used to create data visualizations with a drag-and-drop gesture. It gives you an immediate view of your data without the need to preselect the visualization output. Learn how to create and use [VisBuilder]({{site.url}}{{site.baseurl}}/dashboards/visbuilder/) in Dashboards.

Reviewer: Should this be something like "Reference [VisBuilder]({{site.url}}{{site.baseurl}}/dashboards/visbuilder/) to learn how to create and use VisBuilder in Dashboards." ?

## Vega

<img src="{{site.url}}{{site.baseurl}}/images/vega-1.png" alt="Example Vega visualization with JSON specification in OpenSearch Dashboards" width="60%" height="30%">

> [Vega](https://vega.github.io/vega/) and [Vega-Lite](https://vega.github.io/vega-lite/) are open-source, declarative language visualization grammars for creating, sharing, and saving interactive data visualizations. Vega visualizations in OpenSearch u flexibility to visualize multidimensional data using a layered approach to build and manipulate visualizations in a structured manner.

Reviewer 2: typo? "Vega visualizaitions in OpenSearch u flexibility..."

> [Vega](https://vega.github.io/vega/) and [Vega-Lite](https://vega.github.io/vega-lite/) are open-source, declarative language visualization grammars for creating, sharing, and saving interactive data visualizations. Vega visualizations in OpenSearch gives you flexibility to visualize multidimensional data using a layered approach to build and manipulate visualizations in a structured manner.

Reviewer: Same as comment above.

---
layout: default
title: Visualize
nav_order: 4
has_children: true
---

# Visualize

> Data visualization is representing your data in a visual context. By visualizing your data, you translate complex, high-volume, or numerical data into a visual representation that is easier to process.

Reviewer 3: This description is a little over-complicated for an intro on visualization. I would remove this entire paragraph.

Author: Removed and revised.

> OpenSearch Dashboards gives you data visualization tools to improve and automate the visual communication process. Using visual elements like charts, graphs, or maps to represent data gives you the tools to advance business intelligence and support data-driven decision-making and strategic planning.

Reviewer 3: OpenSearch Dashboards gives you data visualization tools to improve and automate the creation of visual elements in which to display data. Using visual...

> # Visualization types in OpenSearch Dashboards

Reviewer 4: It might be worth it to separate the built in visualization types i.e. Bar, Line etc from the ones that come from plugins e.g. Gantt. Maps will soon be (or is already) a part of this list.

Author: @ashwin-pc Which types are built-in and which types are plugins? Where do I find this information?

Reviewer 5: There's no good resource on that without reading the code, unfortunately. The only one on this list is `Gantt charts`.

Author: I'm working on subpages for each type or tool. Are you all okay with keeping this page higher level and then addressing these distinctions in the subpage content?

> # Visualization types in OpenSearch Dashboards

Author: Revised to add. My oversight.

> # Visualization types in OpenSearch Dashboards

Author: Revised to include these types.

> # Visualization types in OpenSearch Dashboards

Author: @joshuarrrr Please review to confirm the additions accurately reflect the information we want to include at this time.

> Dashboards has several visualization types to support your data analysis needs. The following sections give you an overview of the visualization types in Dashboards and their common use cases.

Reviewer 6: nit, but do you also want to mention the visualization type names within each section? 

In the UI they can select these types: Area, Vertical bar, Horizontal bar, etc. . . 
or maybe this is already covered in the how-to section "Create visualization"

Author: We are grouping the types as area, line, and bar for this intro. We'll have subpages for each specific type, so we'll go into detail on the subpages.

> ## Area charts:  Basic and stacked

Reviewer 6: remove extra space after colon

> Area charts depict changes over time, and they are commonly used to show trends. Area charts more efficiently identify patterns in log data, such as sales data for a time range and trends over that time. Reference [Graphs and charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/graphs-and-charts) to learn more about how to create and use them in Dashboards.

Reviewer 6: just a suggestion

"To learn more about how to create and use area charts, see [Graphs and charts](url)."

 <img src="{{site.url}}{{site.baseurl}}/images/area-chart-1.png" alt="Example area chart in OpenSearch Dashboards" height="90%">

## Bar charts

Bar charts, vertical or horizontal, compare categorical data and demonstrate changes of a variable over a period of time. Reference [Graphs and charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/graphs-and-charts) to learn more about how to create and use them in Dashboards.

<table style="table-layout: fixed ; width: 100%;">
<tbody>
<tr style="text-align: center; vertical-align:center;">
<td><img src="{{site.url}}{{site.baseurl}}/images/bar-chart-1.png" alt="Example vertical bar chart in OpenSearch Dashboards" height="100"></td>
<td><img src="{{site.url}}{{site.baseurl}}/images/bar-horizontal-1.png" alt="Example horizontal bar chart in OpenSearch Dashboards" height="100"></td>
</tr>
<tr style="text-align: center; vertical-align:top; font-weight: bold; color: rgb(0,59,92)">
<td>Vertical bar chart</td>
<td>Horizontal bar chart</td>
</tr>
</tbody>
</table>

## Data tables

> Data tables, or tables, show your raw data in tabular form. 

Reviewer 6: optional: use viz type name

"Use the **Data Table** visualization to show your raw data in. . ."

although maybe the viz type names are obvious enough not to call them out?

Reference [Data tables]({{site.url}}{{site.baseurl}}/dashboards/visualize/tables) to learn how to create and use them in Dashboards.

<img src="{{site.url}}{{site.baseurl}}/images/data-table-1.png" alt="Example data table in OpenSearch Dashboards" height="100">

## Gantt chart

Gantt charts show the start, end, and duration of unique events in a sequence. Gantt charts are useful in trace analytics, telemetry, and anomaly detection use cases where you want to understand interactions and dependencies between various events in a schedule. Reference [Gantt charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/gantt/) to learn how to create and use them in Dashboards.

<img src="{{site.url}}{{site.baseurl}}/images/gantt-chart.png" alt="Example Gantt chart in OpenSearch Dashboards" height="100">

> ## Gauge charts

Author: Revised to include it.

> A gauge chart displays how much there is of the thing you are measuring. In a gauge chart, this measurement can exist alone or in relation to another measurement, such as comparing actual sales to the sales goal. Reference [Gauge charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/gauge) to learn how to create and use them in Dashboards.

Reviewer 6: maybe give additional sentence at the end to describe it visually?

"Gauge charts look similar to an analog speedometer that reads left-to-right from zero.

Author: great addition!

<img src="{{site.url}}{{site.baseurl}}/images/gauge-1.png" alt="Example gauge chart in OpenSearch Dashboards" width="90%">

## Heat maps

A heat map is a view of a histogram (a graphical representation of the distribution of numerical data) over time. Instead of using bar height as a representation of frequency, as with a histogram, heat maps use cells, or coloring a cell proportional to the number of values. Reference [Heat maps]({{site.url}}{{site.baseurl}}/dashboards/visualize/heat-maps) to learn how to create and use them in Dashboards.

<img src="{{site.url}}{{site.baseurl}}/images/heat-map-1.png" alt="Example heat map in OpenSearch Dashboards" height="65">

## Line graphs

> Line charts compare changes in measure values over period of time, such as gross sales by month or gross sales and net sales by month. Reference [Graphs and charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/graphs-and-charts) to learn more about how to create and use them in Dashboards.

Reviewer 6: ..."a period of time"

<img src="{{site.url}}{{site.baseurl}}/images/line-1.png" alt="Example line graph in OpenSearch Dashboards" height="100">

> ## Metric charts

Reviewer 5: "metric charts" isn't quite right - they're not really charts.

Author: Revised. Is "Metric values" the more accurate wording?

Reviewer 5: "metric values" is better. Yeah, some tools call these "Big Numbers" or something like that.

Metric charts show a numerical value, such as a key performance indicator (KPI), to visualize a comparison between a key value and its target value. Metric charts display a value comparison, the two values being compared, and a progress bar. Reference [Metric charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/metric-charts) to learn how to create and use them in Dashboards.

<img src="{{site.url}}{{site.baseurl}}/images/metric-chart-1.png" alt="Example metric chart in OpenSearch Dashboards" height="100">

## Maps
### Coordinate maps

Coordinate maps show location-based data on a map. Use coordinate maps to visualize GPS data (latitude and longitude coordinates) on a map. For information about OpenSearch-supported coordinate field types, see [Geographic field types]({{site.url}}{{site.baseurl}}/opensearch/supported-field-types/geo-shape/) and [Cartesian field types]({{site.url}}{{site.baseurl}}/opensearch/supported-field-types/xy/).

### Region maps

> Region maps show patterns and trends across geographic locations. A region map is one of the basemaps in Dashboards. For information about creating custom vector maps in Dashboards, see [Region maps visualizations](https://opensearch.org/docs/latest/dashboards/geojson-regionmaps/).

Reviewer 6: should base maps be one word?

Reference [Maps]({{site.url}}{{site.baseurl}}/dashboards/visualize/maps/) to learn how to create and use them in Dashboards. 

<img src="{{site.url}}{{site.baseurl}}/images/map-1.png" alt="Example coordinate map in OpenSearch Dashboards" height="100">

## Pie charts

> Pie charts compare values for items in a dimension, such as  a percentage of a total amount. Reference [Pie charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/pie-charts/) to learn how to create and use them in Dashboards.

Reviewer 6: rmv extra space "...as a percentage"

<img src="{{site.url}}{{site.baseurl}}/images/pie-1.png" alt="Example pie chart in OpenSearch Dashboards" height="100">

> ## TSVB

Author: Revised each tool section. Please review changes to verify accuracy.

> Time-series visual builder provides a detailed time series visualization. For example, you can use TSVB to show data over time, such as flights by status over time or flight delays by delay type over time. Reference [TSVB]({{site.url}}{{site.baseurl}}/dashboards/visualize/TSVB/) to learn how to create and use this set of visualization types in Dashboards.

Reviewer 6: The time-series visual builder (TSVB). . .

Reviewer 3: +1

Author: Revised

<img src="{{site.url}}{{site.baseurl}}/images/TSVB-1.png" alt="Example TSVB in OpenSearch Dashboards" height=100>

## Tag cloud

Tag (or word) clouds are a way to display how often a word is used in relation to other words in a dataset. The best use for this type of visual is to show word or phrase frequency. Learn how to create and use [Tag clouds]({{site.url}}{{site.baseurl}}/dashboards/visualize/tag-clouds/) in Dashboards.

<img src="{{site.url}}{{site.baseurl}}/images/word-cloud-1.png" alt="Example Tag cloud in OpenSearch Dashboards" height="100">

> ## Timeline

Reviewer 4: Should we call out Timeline, VisBuilder and Vega out separately since they are less of a visualization type and more like a tool that offers a different experience for creating visualizations?

Author: Revised each of these sections to clarify they are tools. Please re-read these sections to confirm the changes are accurate descriptions.

Author: Revised each to clarify they are tools, not types. Please review to verify the changes are accurate descriptions.

> Timelines  show the occurrence of events in chronological order, allowing  you to see when events occur and how they change over time. Reference [Timelines]({{site.url}}{{site.baseurl}}/dashboards/visualize/timeline/) to learn how to create and use timelines in Dashboards.

Reviewer 6: rmv extra spaces  "Timelines show" and "allowing you"

> Timelines  show the occurrence of events in chronological order, allowing  you to see when events occur and how they change over time. Reference [Timelines]({{site.url}}{{site.baseurl}}/dashboards/visualize/timeline/) to learn how to create and use timelines in Dashboards.

Author: Revised. Please review to verify the changes are accurate.

---
layout: default
title: Visualize
nav_order: 4
has_children: true
---

# Visualize

> By visualizing your data, you translate complex, high-volume, or numerical data into a visual representation that is easier to process. OpenSearch Dashboards gives you data visualization tools to improve and automate the visual communication process. Using visual elements like charts, graphs, or maps to represent data gives you the tools to advance business intelligence and support data-driven decision-making and strategic planning.

Author: Revised second use to read "...to represent data advances business intelligence and supports...."

> By visualizing your data, you translate complex, high-volume, or numerical data into a visual representation that is easier to process. OpenSearch Dashboards gives you data visualization tools to improve and automate the visual communication process. Using visual elements like charts, graphs, or maps to represent data gives you the tools to advance business intelligence and support data-driven decision-making and strategic planning.

Reviewer 7: I meant that "gives you data visualization tools" and "gives you the tools" are a bit similar in construct, but it's not a big deal :-).

> By visualizing your data, you translate complex, high-volume, or numerical data into a visual representation that is easier to process. OpenSearch Dashboards gives you data visualization tools to improve and automate the visual communication process. Using visual elements like charts, graphs, or maps to represent data gives you the tools to advance business intelligence and support data-driven decision-making and strategic planning.

Author: How about this? By using visual elements like charts, graphs, or maps to represent data, you can advance business intelligence and support data-driven decision-making and strategic planning."

# Visualization types in OpenSearch Dashboards

Dashboards has several visualization types to support your data analysis needs. The following sections give you an overview of the visualization types in Dashboards and their common use cases.

## Area charts

Area charts depict changes over time, and they are commonly used to show trends. Area charts more efficiently identify patterns in log data, such as sales data for a time range and trends over that time. Reference [Graphs and charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/graphs-and-charts) to learn more about how to create and use them in Dashboards.

 <img src="{{site.url}}{{site.baseurl}}/images/area-chart-1.png" alt="Example area chart in OpenSearch Dashboards" height="90%">

## Bar charts

> Bar charts, vertical or horizontal, compare categorical data and demonstrate changes of a variable over a period of time. Reference [Graphs and charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/graphs-and-charts) to learn more about how to create and use them in Dashboards.

Reviewer 7: ```suggestion
Bar charts, vertical or horizontal, compare categorical data and depict changes of a variable over a period of time. Reference [Graphs and charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/graphs-and-charts) to learn more about how to create and use them in Dashboards.
```

<table style="table-layout: fixed ; width: 100%;">
<tbody>
<tr style="text-align: center; vertical-align:center;">
<td><img src="{{site.url}}{{site.baseurl}}/images/bar-chart-1.png" alt="Example vertical bar chart in OpenSearch Dashboards" height="100"></td>
<td><img src="{{site.url}}{{site.baseurl}}/images/bar-horizontal-1.png" alt="Example horizontal bar chart in OpenSearch Dashboards" height="100"></td>
</tr>
<tr style="text-align: center; vertical-align:top; font-weight: bold; color: rgb(0,59,92)">
<td>Vertical bar chart</td>
<td>Horizontal bar chart</td>
</tr>
</tbody>
</table>

## Controls

> Controls is a panel, instead of a visualization type, added to a dashboard to filter data. Controls gives users the capability to add interactive inputs to a dashboard. You can create two types of controls in Dashboards: **Options list** and **Range slider**. **Options list** is a drop-down options list that allows filtering of data by a terms aggregation, such as `machine.os.keyword`. **Range slider** allows filtering within specified value ranges, such as `hour_of_day`.  

Reviewer 7: ```suggestion
Controls is a panel, instead of a visualization type, added to a dashboard to filter data. Controls gives users the capability to add interactive inputs to a dashboard. You can create two types of controls in Dashboards: **Options list** and **Range slider**. **Options list** is a dropdown options list that allows filtering of data by a terms aggregation, such as `machine.os.keyword`. **Range slider** allows filtering within specified value ranges, such as `hour_of_day`.  
```

<img src="{{site.url}}{{site.baseurl}}/images/controls-1.png" alt="Example visualization using controls to filter data in OpenSearch Dashboards" height="100">

## Data tables

Data tables, or tables, show your raw data in tabular form. 
Reference [Data tables]({{site.url}}{{site.baseurl}}/dashboards/visualize/tables) to learn how to create and use them in Dashboards.

<img src="{{site.url}}{{site.baseurl}}/images/data-table-1.png" alt="Example data table in OpenSearch Dashboards" height="100">

## Gantt charts

> Gantt charts show the start, end, and duration of unique events in a sequence. Gantt charts are useful in trace analytics, telemetry, and anomaly detection use cases where you want to understand interactions and dependencies between various events in a schedule. **Gantt chart** is currently a plug-in, instead of built-in, visualization type in Dashboards. Reference [Gantt charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/gantt/) to learn how to create and use them in Dashboards.

Reviewer 7: ```suggestion
Gantt charts show the start, end, and duration of unique events in a sequence. Gantt charts are useful in trace analytics, telemetry, and anomaly detection use cases where you want to understand interactions and dependencies between various events in a schedule. **Gantt chart** is currently a plugin, instead of built-in, visualization type in Dashboards. Reference [Gantt charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/gantt/) to learn how to create and use them in Dashboards.
```

<img src="{{site.url}}{{site.baseurl}}/images/gantt-chart.png" alt="Example Gantt chart in OpenSearch Dashboards" height="100">

## Gauge charts

> Gauge charts look similar to an analog speedometer that reads left-to-right from zero. They display how much there is of the thing you are measuring, and this measurement can exist alone or in relation to another measurement, such as tracking performance against benchmarks or goals. Reference [Gauge charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/gauge) to learn how to create and use them in Dashboards.

Reviewer 7: ```suggestion
Gauge charts look similar to an analog speedometer that reads left to right from zero. They display how much there is of the thing you are measuring, and this measurement can exist alone or in relation to another measurement, such as tracking performance against benchmarks or goals. Reference [Gauge charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/gauge) to learn how to create and use them in Dashboards.
```

<img src="{{site.url}}{{site.baseurl}}/images/gauge-1.png" alt="Example gauge chart in OpenSearch Dashboards" width="90%">

## Heat maps

> A heat map is a view of a histogram (a graphical representation of the distribution of numerical data) over time. Instead of using bar height as a representation of frequency, as with a histogram, heat maps use cells, or coloring a cell proportional to the number of values. Reference [Heat maps]({{site.url}}{{site.baseurl}}/dashboards/visualize/heat-maps) to learn how to create and use them in Dashboards.

Reviewer 7: The second sentence needs a bit of revision after "histogram". Do we mean something like "...as with a histogram, heat maps show X by coloring cells to..."?

Author: Revised to read: "...heat maps display data in a tabular form using colors to differentiate where values fall in a range."

<img src="{{site.url}}{{site.baseurl}}/images/heat-map-1.png" alt="Example heat map in OpenSearch Dashboards" height="65">

## Line graphs

> Line charts compare changes in measure values over a period of time, such as gross sales by month or gross sales and net sales by month. Reference [Graphs and charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/graphs-and-charts) to learn more about how to create and use them in Dashboards.

Reviewer 7: The heading uses "graphs", but we use "charts" here. Should "measure" be "measured"?

<img src="{{site.url}}{{site.baseurl}}/images/line-1.png" alt="Example line graph in OpenSearch Dashboards" height="100">

## Maps

> ### Coordinate maps

Reviewer 7: Text needs to separate the headings so that they are not "stacked."

Author: You can create two types of maps in Dashboards: Coordinate maps and Region maps. Coordinate maps show the difference between data values for each location by size. Region maps show the difference between data values for each location by varying shades of color.

Coordinate maps show location-based data on a map. Use coordinate maps to visualize GPS data (latitude and longitude coordinates) on a map. For information about OpenSearch-supported coordinate field types, see [Geographic field types]({{site.url}}{{site.baseurl}}/opensearch/supported-field-types/geo-shape/) and [Cartesian field types]({{site.url}}{{site.baseurl}}/opensearch/supported-field-types/xy/).

### Region maps

> Region maps show patterns and trends across geographic locations. A region map is one of the basemaps in Dashboards. For information about creating custom vector maps in Dashboards, see [Region maps visualizations](https://opensearch.org/docs/latest/dashboards/geojson-regionmaps/).

Reviewer 7: ```suggestion
Region maps show patterns and trends across geographic locations. A region map is one of the basemaps in Dashboards. For information about creating custom vector maps in Dashboards, see [Region map visualizations](https://opensearch.org/docs/latest/dashboards/geojson-regionmaps/).
```

> Reference [Maps]({{site.url}}{{site.baseurl}}/dashboards/visualize/maps/) to learn how to create and use them in Dashboards. 

Reviewer 7: ```suggestion
See [Maps]({{site.url}}{{site.baseurl}}/dashboards/visualize/maps/) to learn how to create and use them in Dashboards. 
```

Reviewer 7: Changed because my brain read "Reference maps" as a noun.

<img src="{{site.url}}{{site.baseurl}}/images/map-1.png" alt="Example coordinate map in OpenSearch Dashboards" height="100">

## Markdown

Markdown is a the markup language used in Dashboards to provide context to your data visualizations. Using Markdown, you can display information and instructions along with the visualization. Reference [Markdown]({{site.url}}{{site.baseurl}}/dashboards/visualize/markdown/) to learn how to use this markup language with visualizations in Dashboards.

<img src="{{site.url}}{{site.baseurl}}/images/markdown-1.png" alt="Example coordinate map in OpenSearch Dashboards" height="100">

## Metric values

> Metric values, or number charts, show a numerical value for a specific period,  such as total sales year to date, and you can add a trend indicator, such as sales from the last period,  to compare the current data to data from another period. Metric values display the percentage or value change between . Reference [Metric charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/metric-charts) to learn how to create and use them in Dashboards.

Reviewer 7: ```suggestion
Metric values, or number charts, show a numerical value for a specific period, such as total sales year to date, and you can add a trend indicator, such as sales from the last period, to compare the current data to data from another period. Metric values display the percentage or value change between . Reference [Metric charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/metric-charts) to learn how to create and use them in Dashboards.
```

Reviewer 7: The penultimate sentence is incomplete.

Author: Revised to simplify the first two sentences. "Metric values, or number charts, compare values in different measures. For example, you can create a metrics visualization to compare two values, such as actual sales compared to sales goals."

## Pie charts

Pie charts compare values for items in a dimension, such as a percentage of a total amount. Reference [Pie charts]({{site.url}}{{site.baseurl}}/dashboards/visualize/pie-charts/) to learn how to create and use them in Dashboards.

<img src="{{site.url}}{{site.baseurl}}/images/pie-1.png" alt="Example pie chart in OpenSearch Dashboards" height="100">

## TSVB

The time-series visual builder (TSVB) is a data visualization tool in Dashboards used to create detailed time-series visualizations. For example, you can use TSVB to build visualizations that show data over time, such as flights by status over time or flight delays by delay type over time. Currently, TSVB can be used to create the following Dashboards visualization types: Area, Line, Metric, Gauge, Markdown, and Data Table. Reference [TSVB]({{site.url}}{{site.baseurl}}/dashboards/visualize/TSVB/) to learn how to create and use this set of visualization types in Dashboards.

<img src="{{site.url}}{{site.baseurl}}/images/TSVB-1.png" alt="Example TSVB in OpenSearch Dashboards" height=100>

## Tag cloud

> Tag (or word) clouds are a way to display how often a word is used in relation to other words in a dataset. The best use for this type of visual is to show word or phrase frequency. Learn how to create and use [Tag clouds]({{site.url}}{{site.baseurl}}/dashboards/visualize/tag-clouds/) in Dashboards.

Reviewer 7: I would rephrase the last sentence to match the structure of the previous sentences that refer the reader to documentation.

Author: Done

<img src="{{site.url}}{{site.baseurl}}/images/word-cloud-1.png" alt="Example Tag cloud in OpenSearch Dashboards" height="100">

## Timeline

> Timeline is a data visualization tool in Dashboards that you can use to create time-series visualizations. Currently, Timeline can be used to create the following Dashboards visualization types: Area and Line. Reference [Timelines]({{site.url}}{{site.baseurl}}/dashboards/visualize/timeline/) to learn how to create and use timelines in Dashboards.

Reviewer 7: ```suggestion
Timeline is a data visualization tool in Dashboards that you can use to create time-series visualizations. Currently, Timeline can be used to create the following Dashboards visualization types: Area and Line. Reference [Timelines]({{site.url}}{{site.baseurl}}/dashboards/visualize/timeline/) to learn how to create and use them in Dashboards.
```

<img src="{{site.url}}{{site.baseurl}}/images/timeline-1.png" alt="Example Timeline in OpenSearch Dashboards" height="100">

## VisBuilder

> VisBuilder is a drag-and-drop data visualization tool in Dashboards. It gives you an immediate view of your data without the need to preselect the visualization type output. Currently, VisBuilder can be used to create the following Dashboards visualization types: Area, Bar, Line, Metric, and Data Table. Reference [VisBuilder]({{site.url}}{{site.baseurl}}/dashboards/visbuilder/) to learn how to create and use drag-and-drop visualizations in Dashboards.

Reviewer 4: ```suggestion
VisBuilder is a drag-and-drop data visualization tool in Dashboards. It gives you an immediate view of your data without the need to preselect the datasource or visualization type output. Currently, VisBuilder can be used to create the following Dashboards visualization types: Area, Bar, Line, Metric, and Data Table. Reference [VisBuilder]({{site.url}}{{site.baseurl}}/dashboards/visbuilder/) to learn how to create and use drag-and-drop visualizations in Dashboards.
```

> VisBuilder is a drag-and-drop data visualization tool in Dashboards. It gives you an immediate view of your data without the need to preselect the datasource or visualization type output. Currently, VisBuilder can be used to create the following Dashboards visualization types: Area, Bar, Line, Metric, and Data Table. Reference [VisBuilder]({{site.url}}{{site.baseurl}}/dashboards/visbuilder/) to learn how to create and use drag-and-drop visualizations in Dashboards.

Reviewer 7: Confirm that "datasource" shouldn't be "data source".

Author: Confirmed it's "data source" in Dashboards. Revised.

<img src="{{site.url}}{{site.baseurl}}/images/drag-drop-generated-viz.png" alt="Example Timeline in OpenSearch Dashboards" height="100">

## Vega

> [Vega](https://vega.github.io/vega/) and [Vega-Lite](https://vega.github.io/vega-lite/) are open-source, declarative language visualization grammars for creating, sharing, and saving interactive data visualizations. Vega visualizations in OpenSearch gives you flexibility to visualize multidimensional data using a layered approach to build and manipulate visualizations in a structured manner. Vega can be used to create customized visualizations using any Dashboards visualization type.

Reviewer 7: ```suggestion
[Vega](https://vega.github.io/vega/) and [Vega-Lite](https://vega.github.io/vega-lite/) are open-source, declarative language visualization grammars for creating, sharing, and saving interactive data visualizations. Vega visualizations give you the flexibility to visualize multidimensional data using a layered approach in order to build and manipulate visualizations in a structured manner. Vega can be used to create customized visualizations using any Dashboards visualization type.
```
